File: Server/go-server/test/user-client/user_client.go
package uclient

import (
	"context"

	skp "github.com/sirfrank96/go-server/sports-keypoints-proto"
	//testutil "github.com/sirfrank96/go-server/test/test-util"

	"google.golang.org/grpc"
	"google.golang.org/grpc/credentials/insecure"
)

//TODO: RETURN ERRORS INSTEAD OF LOG.FATALF

// Middle arg is a close function, should be called by calling function
func InitUserServiceGrpcClient(serveraddr string) (skp.UserServiceClient, func() error, error) {
	// Set up a connection to the cv_api server.
	conn, err := grpc.NewClient(serveraddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
	if err != nil {
		return nil, conn.Close, err
	}
	// Init ComputerVisionGolf grpc client
	uclient := skp.NewUserServiceClient(conn)
	return uclient, conn.Close, nil
}

func CreateUser(ctx context.Context, uclient skp.UserServiceClient, userName string, password string, email string) (*skp.CreateUserResponse, error) {
	request := &skp.CreateUserRequest{
		UserName: userName,
		Password: password,
		Email:    email,
	}
	return uclient.CreateUser(ctx, request)
}

func RegisterUser(ctx context.Context, uclient skp.UserServiceClient, userName string, password string) (*skp.RegisterUserResponse, error) {
	request := &skp.RegisterUserRequest{
		UserName: userName,
		Password: password,
	}
	return uclient.RegisterUser(ctx, request)
}

File: clients/python-golf-client/src/user_client.py
import user_pb2
import user_pb2_grpc

#client stub functions for user grpc service
class UserClient():
    def __init__(self, channel):
        self.stub = user_pb2_grpc.UserServiceStub(channel)

    def create_user(self, username, password, email):
        request = user_pb2.CreateUserRequest(user_name=username, password=password, email=email)
        return self.stub.CreateUser(request)
    
    def register_user(self, username, password):
        request = user_pb2.RegisterUserRequest(user_name=username, password=password)
        return self.stub.RegisterUser(request)

    def read_user(self, session_token):
        request = user_pb2.ReadUserRequest(session_token=session_token)
        return self.stub.ReadUser(request)

    def update_user(self, session_token, username, password, email):
        request = user_pb2.UpdateUserRequest(session_token=session_token, user_name=username, password=password, email=email)
        return self.stub.UpdateUser(request)

    def delete_user(self, session_token):
        request = user_pb2.DeleteUserRequest(session_token=session_token)
        return self.stub.DeleteUser(request)
    